main/zoe/api/zoe-contract-facet.md

```diff
@@ -259,11 +259,19 @@ to manipulate the offer. The queries and operations are as follows:
         Price: moolaAmountMath.make(9)
       }
       ```
- - `exit()`
+ - `exit(completion)`
    - Returns: `void`
    - Causes the `seat` to exit, concluding its existence. All `payouts`, if any,
-     are made, and the `seat` object can no longer interact with the contract. .
- - `kickOut(msg)` (`msg` is optional)
+     are made, and the `seat` object can no longer interact with the contract.
+     The `completion` argument is usually a string, but this is not required. **tyg todo:
+     What else can it be and what is it used for/does it do?**
+ - `fail(msg)` (`msg` is optional)
+   - Returns: `void`
+   - The `seat` exits, displaying the `msg` string, if there is one, on the console.
+     This is equivalent to exiting, except that `exit` is for a successful transaction while
+     `fail()` aborts the transaction attempt and signals an error. The contract
```

Comment:
```suggestion
     `fail()` signals an error. The contract
```

The only difference is the way it signals termination. In both cases, other functions called in the same turn will complete. in both cases the seat gets its final allocation, and in both cases there is no further interaction with the contract.

Reply:
Yup, I agree with Chris. We need to take out the term `transaction` - it implies that there's such a thing, but nothing like it actually exists. How about:

```suggestion
     
     This is equivalent to exiting, except that `exit` is successful while `fail()` signals an error occurred while processing the offer.
```